### ISSUE TITLE
Expose the cluster name as part of the credentials environment.

### ISSUE THREAD

**Issue by BrowncoatShadow** [getcarina/carina#113]
It would be very useful to have the cluster name be exposed to an environment variable as part of the credentials evaluation. This would be analogous to docker machine's `DOCKER_MACHINE_NAME` environment variable that is exposed as part of their `env` subcommand.

Currently, there is no way to check what cluster you are pointed to unless you can recognize it based on the output of `docker info` or the deployed containers. Being able to explicitly know would be helpful, especially for scripting.

Something like:
```sh
$ eval "$(carina env mycluster)"
$ echo $CARINA_CLUSTER_NAME
mycluster 
```

**Comment by BrowncoatShadow** [comment]
Here is an excerpt from a wrapper script based on docker machine's wrapper script, which adds a `use` subcommand, that could make use of this functionality.

```sh
...
eval "$(carina env "$@")"
echo "Active cluster: ${CARINA_CLUSTER_NAME}"

# Unless disabled, check for and try to use DVM
if [[ -z ${CARINA_NO_DVM} ]] && [[ "$(type -p dvm)" ]]; then
  dvm use
fi
...
```

I would also be happy with this functionality being added natively to the client. 😉

**Comment by carolynvs** [comment]
Are you also suggesting that carina try to call `dvm use` on your behalf if dvm is around as part of `carina use`? Or are you just looking for the env var?

**Comment by BrowncoatShadow** [comment]
@carolynvs, I think the environment variable is the most important part, it goes beyond just my particular use case. That being said, I do think a `carina use` would really help with workflow. In my experience with Docker Machine, it is incredibly helpful. Especially when working with multiple targets.

As for automatically calling `dvm use`, it would be my preference to do so. It would boil down the process to target an existing cluster to one command (one that is also simpler than `eval`'ing a command substitution). I think *most* of the time the user is going to have DVM installed and need to use it to access their cluster anyway. Especially if they have some older clusters that may not all be using the same version.

In my example I have the `dvm use` part enabled by default, with the option to disable it with an environment variable. I could go either way with on or off by default. On by default just eliminates a setup step.

**Comment by carolynvs** [comment]
Cool, adding the env var is a bit more straightforward so I'll tackle that for now and we can go from there.

For the carina cli to add env vars to the current shell session would require a lot more plumbing work (don't know if you've ever peeked under the hood at how dvm does its magic). Not saying that it isn't worth looking at but baby steps. 😄

**Comment by carolynvs** [comment]
@BrowncoatShadow Want to take a look at https://github.com/getcarina/libcarina/pull/18 and try out the prerelease binaries? Let me know if that works the way you are expecting and if so, I'll do a release.

**Comment by carolynvs** [comment]
Sorry for the delay in publishing the release! This is now available in [v1.5.0](https://github.com/getcarina/carina/releases/tag/v1.5.0).
